orchestra/todos/api.py

```diff
@@ -17,17 +21,50 @@ def add_todolist_template(todolist_template_slug, task_id):
         template=todolist_template
     )
     root_todo.save()
+
+    cond_props = {}
+    path = todolist_template.conditional_property_function.get(
+        'path', None)
+    if path:
+        try:
+            get_cond_props = locate(path)
+            cond_props = get_cond_props(task.project)
+        except Exception:
+            logger.exception('Invalid condition function path.')
     for template_todo in template_todos:
-        _add_template_todo(template_todo, todolist_template, root_todo, task)
+        _add_template_todo(
+            template_todo, todolist_template, root_todo, task, cond_props)
 
 
-def _add_template_todo(template_todo, todolist_template, parent_todo, task):
-    todo = Todo(
-        task=task,
-        description=template_todo['description'],
-        template=todolist_template,
-        parent_todo=parent_todo
-    )
-    todo.save()
-    for template_todo_item in template_todo.get('items', []):
-        _add_template_todo(template_todo_item, todolist_template, todo, task)
+def _to_exclude(props, conditions=[]):
+    for condition in conditions:
+        for prop, predicate in condition.items():
+            current_value = props.get(prop)
+            compared_to_value = predicate['value']
+            if predicate['operator'] == '=':
+                return current_value == compared_to_value
+            elif predicate['operator'] == '!=':
+                return current_value != compared_to_value
+    return False
+
+
+def _add_template_todo(
+        template_todo, todolist_template,
+        parent_todo, task, conditional_props):
+    remove = _to_exclude(conditional_props, template_todo.get('remove_if', []))
+    if not remove:
+        to_skip = _to_exclude(
+                conditional_props, template_todo.get('skip_if', []))
+        skipped_datetime = timezone.now() if to_skip else None
+        todo = Todo(
+            task=task,
+            description=template_todo['description'],
+            template=todolist_template,
+            parent_todo=parent_todo,
+            skipped_datetime=skipped_datetime
```

Comment:
To propogate skipping from parents to descendents.
```python
skipped_datetime = timezone.now() if parent_todo.skipped_datetime or to_skip else None
```
We may even want to skip evaluation of `to_skip` if parent_todo.skipped_datetime is not Null. That may save considerable computation time.